photon-ml/src/main/scala/com/linkedin/photon/ml/data/KeyValueScore.scala

```diff
@@ -63,7 +63,12 @@ protected[ml] class KeyValueScore(val scores: RDD[(Long, Double)]) extends RDDLi
     val addedScores =
       this.scores
         .fullOuterJoin(that.scores)
-        .mapValues { case (thisScore, thatScore) => thisScore.getOrElse(0.0) + thatScore.getOrElse(0.0) }
+        .mapValues { case (thisScore, thatScore) =>
+          require(!thisScore.isEmpty || !thatScore.isEmpty, s"Should not have reached this state. A full outer join " +
+            s"should yield a result that has at least one of $thisScore or $thatScore non-empty")
+          thisScore.getOrElse(thatScore.get.getZeroScoreDatum()) +
+            thatScore.getOrElse(thisScore.get.getZeroScoreDatum())
```

Comment:
With a full outer join, one of the two values must be there. I think this `require` is a defensive move since we're calling `get` on the options in L69-70. As @ashelkovnykov suggested, a match/case block would be a more natural and idiomatic way of dealing with this.